domain/lib/src/model/offline_mode/document_offline.dart

```diff
@@ -0,0 +1,116 @@
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+//
+
+import 'package:domain/domain.dart';
+import 'package:domain/src/model/document/document_id.dart';
+import 'package:equatable/equatable.dart';
+import 'package:http_parser/http_parser.dart';
+
+class DocumentOffline extends Equatable {
+  final DocumentId documentId;
+  final String description;
+  final DateTime creationDate;
+  final DateTime modificationDate;
+  final DateTime expirationDate;
+  final bool ciphered;
+  final String name;
+  final int size;
+  final String sha256sum;
+  final bool hasThumbnail;
+  final int shared;
+  final MediaType mediaType;
+  final String localPath;
+
+  DocumentOffline(
+    this.documentId,
+    this.description,
+    this.creationDate,
+    this.modificationDate,
+    this.expirationDate,
+    this.ciphered,
+    this.name,
+    this.size,
+    this.sha256sum,
+    this.hasThumbnail,
+    this.shared,
+    this.mediaType,
+    this.localPath
+  );
+
+  Map<String, dynamic> toMap() => {
+    'documentId': documentId != null ? documentId.uuid : null,
+    'description': description,
+    'creationDate': creationDate != null ? creationDate.millisecondsSinceEpoch : null,
+    'modificationDate': modificationDate != null ? modificationDate.millisecondsSinceEpoch : null,
+    'expirationDate': expirationDate != null ? expirationDate.millisecondsSinceEpoch : null,
+    'ciphered': ciphered != null ? (ciphered ? 1 : 0) : null,
+    'name': name,
+    'size': size,
+    'sha256sum': sha256sum,
+    'hasThumbnail': hasThumbnail != null ? (hasThumbnail ? 1 : 0) : null,
+    'shared': shared,
+    'type': mediaType != null ? mediaType.mimeType : null,
+    'localPath': localPath,
+  };
+
+  factory DocumentOffline.fromMap(Map<String, dynamic> json) => DocumentOffline(
+    json['uuid'] != null ? DocumentId(json['uuid'] as String) : null,
+    json['description'] as String,
+    json['creationDate'] != null ? DateTime.fromMillisecondsSinceEpoch(json['creationDate'] as int) : null,
+    json['modificationDate'] != null ? DateTime.fromMillisecondsSinceEpoch(json['modificationDate'] as int) : null,
+    json['expirationDate'] != null ? DateTime.fromMillisecondsSinceEpoch(json['expirationDate'] as int) : null,
+    json['ciphered'] != null ? ((json['ciphered'] as int) == 1 ? true : false) : null,
+    json['name'] as String,
+    json['size'] as int,
+    json['sha256sum'] as String,
+    json['hasThumbnail'] != null ? ((json['hasThumbnail'] as int) == 1 ? true : false) : null,
+    json['shared'] as int,
+    json['type'] != null ? MediaType.parse(json['type'] as String) : null,
+    json['localPath'] as String
+  );
```

Comment:
This is for serialization from database?. No better way for it?

Reply:
I'm looking for an easier conversion library for this json string. But since it's tied to the Sqflite database library, I haven't found a way to optimize it yet.